### PR TITLE
dialects (arith): Add arith.addui_extended operation

### PR DESCRIPTION
**Pull request by math-fehr**
Add `arith.addui_extended` operation.
It works like `arith.addi`, but returns a flag that indicates if the addition overflowed.

### CODE REVIEW

File: tests/filecheck/dialects/arith/arith_ops.mlir
Comment (superlopuh) [inline tests/filecheck/dialects/arith/arith_ops.mlir:169]:
```suggestion
  %sum, %carry = arith.addui_extended %lhsi32, %rhsi32 : i32, i1
  %sum_index, %carry_index = arith.addui_extended %lhsi64, %rhsi64 : i64, i1
```
Shouldn't these be generic?

Comment (math-fehr) [reply tests/filecheck/dialects/arith/arith_ops.mlir:169]: Half of this file has custom syntax.
Unsure then what we should have there?

Comment (superlopuh) [reply tests/filecheck/dialects/arith/arith_ops.mlir:169]: I personally don't think we need checks for generic syntax but I don't feel strongly about any of this, just thought I'd bring it up to make sure you're aware

**Comment by tobiasgrosser** [comment]
@math-fehr, should we merge this?

**Review by AntonLydike — APPROVED** [review]
Nice stuff! :rocket:

**Comment by math-fehr** [comment]
> @math-fehr, should we merge this?

I'm basing on another branch, so we can't merge it until the other is merged.

**Pull request merged**
